airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -173,7 +173,8 @@ def get_bucket(self, bucket_name: Optional[str] = None) -> str:
         :return: the bucket object to the bucket name.
         :rtype: boto3.S3.Bucket
         """
-        s3_resource = self.get_resource_type('s3')
+        session, endpoint_url = self._get_credentials()
+        s3_resource = session.client('s3', endpoint_url=endpoint_url, config=self.config, verify=self.verify)
```

Comment:
Same question as the comment on redshift in PostgresHook

Reply:
This might be potentially adjusted to use out of the box default value of `self.get_resource_type()` but i'm not 100% certain of value of `self.resource_type` to verify. I could only guess that its `s3`...

Reply:
Ok lets leave it as is now we can investigate this later